src/scenes/boot-scene.js

```diff
@@ -45,5 +67,64 @@ export default class BootScene extends Phaser.Scene {
     groundLayer.setCollisionByProperty({ collides: true });
     return groundLayer;
   }
-}
 
+  addAllDiamonds(){
+    const map = this.make.tilemap({key: "map"});
+    this.groupDiamond = map.createFromObjects("groupDiamond", "diamond", { key: "diamond" }, this);
+    this.groupDiamond.forEach((diamondSprite) => {
+      this.physics.world.enable(diamondSprite);
+      diamondSprite.body.immovable =true;
+      diamondSprite.body.allowGravity = false;
+      this.physics.add.overlap(this.redBall, diamondSprite, this.collectDiamond, null, this);
+    });
+  }
+
+  collectDiamond(redBall, diamond){
+    this.score ++;
+    this.scoreText.setText("x " + this.score);
+    diamond.destroy();
+  }
+
+  addGameOver(){
```

Comment:
It's a good idea to split our method but you need to have for that. in this case `gameOver` is a functionality itself so you can create some method like `runGameOver` and then include the addGameOver, addGameOverSound and addFameOverMessage into it (also the `this.isGameOver = true;` in line 52). we don't need three different methods to understand what runGameOver method is doing

Reply:
Ok, I created a method called runGameOver that include addGameOver, addGameOverSound,  addGameOverMessage and the "this.isGameOver = true;" and put the method runGameOver inside the if(this.redBall is dead && !this.isGameOver)